File: 2269-find-the-k-beauty-of-a-number/2269-find-the-k-beauty-of-a-number.java
class Solution {
    public int divisorSubstrings(int num, int k) {
        
        String t="";
        String S=Integer.toString(num);
        int c=0;
        int i=0;
        int j=0;
        
        while(j<S.length())
        {
            t+=S.charAt(j);
            if(j-i+1<k){ j++; }
            else if(j-i+1==k)
            {
                int a=Integer.parseInt(t.substring(i,i+k));
                if(a!=0 && num%a==0) { c++; }
                i++;
                j++;
            }
        }
        return c;
    }
}

File: 2269-find-the-k-beauty-of-a-number/2269-find-the-k-beauty-of-a-number.cpp
class Solution {
public:
    int divisorSubstrings(int num, int k) {
        int result=0;
        string s=to_string(num);
        int n=s.size();
        int left=0; 
        int right=k-1;
        string temp=s.substr(left, k);
        int temp1=stoi(temp);
        if(temp1!=0 && num%temp1==0)
        {
            result++;
        }
        left++;
        right++;
        while(right<n)
        {
            temp=s.substr(left,k);
            temp1=stoi(temp);
            if(temp1!=0 && num%temp1==0)
            {
                result++;
            }
            left++;
            right++;
        }
        return result;
    }
};